src/main/ruby/core/kernel.rb

```diff
@@ -648,13 +648,14 @@ def printf(*args)
   alias_method :untrusted?, :tainted?
 
   def caller(start = 1, limit = nil)
-    start += 1
-    if limit.nil?
-      args = [start]
-    else
-      args = [start, limit]
-    end
-    Kernel.caller_locations(*args).map(&:to_s)
+    start, limit = if start.is_a? Range
+                     [start.begin, start.size]
+                   elsif limit.nil?
+                     [start, -1]
+                   else
+                     [start, limit]
+                   end
+    Kernel.caller_locations(start + 1, limit).map(&:to_s)
```

Comment:
`limit = -1` is not accepted on MRI:
```
caller_locations(2, -1)
ArgumentError: negative size (-1)
from (pry):9:in `caller_locations'
```

Reply:
ah right, forgot I am in the spec not our tests

Reply:
But this is in our file. Are you objecting to rely on our undocumented behavior? That might indeed not be the best.

Reply:
It's not ideal. I'd prefer to keep it as before and don't pass limit at all if it's not given.
Otherwise a proper fix would be to make caller_locations a primitive, then we don't need this `+ 1` and we can be more liberal about what values are valid arguments.